components/constellation/network_listener.rs

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use hyper::header::Location;
+use ipc_channel::ipc;
+use msg::constellation_msg::PipelineId;
+use net_traits::{FetchMetadata, FetchResponseMsg, FilteredMetadata, NetworkError};
+use net_traits::request::RequestInit;
+use net_traits::response::ResponseInit;
+use script_traits::{ConstellationControlMsg, ScriptMsg};
+
+pub struct NetworkListener {
+    pub res_init: ResponseInit,
+    pub req_init: RequestInit,
+    pub pipeline_id: PipelineId,
+}
+
+impl NetworkListener {
+    pub fn notify(&self, message: Result<FetchMetadata, NetworkError>) {
```

Comment:
I stole most of the names from script/network_listener.rs, since I'm not very good at naming things :P